libs/game/sprite.ts

```diff
@@ -311,19 +394,29 @@ class Sprite extends sprites.BaseSprite {
         return !(this.flags & SpriteFlag.Invisible);
     }
 
+    private recalcSize(): void {
+        this._width = Fx8(this._image.width * this.sx);
+        this._height = Fx8(this._image.height * this.sy);
+        this.resetHitbox();
+    }
+
+    private isScaled(): boolean {
+        return this._sx !== Fx.oneFx8 || this._sy !== Fx.oneFx8;
+    }
+
     //% group="Physics" blockSetVariable="mySprite"
-    //% blockCombine block="width"
+    //% blockCombine block="width" callInDebugger
     get width() {
-        return this._image.width
+        return Fx.toInt(this._width);
```

Comment:
@jwunderl I think a float return type would be more appropriate here, otherwise `s.right - s.left` would not necessarily equal `s.width`. I wonder what you think. Worth "fixing"?

Reply:
Hm, I unless there's a subtlety that I'm missing with regards to the sprite scaling then this value should always be be an integer, no? Or do we count a 15x15 @ 1.5x scale as 22.5x22.5 pixels even though it the pixels will be rounded on render?

It shouldn't hurt as the normal case has it go directly so the image's integer dimensions, so I'd say up to you~